source/hp/dof_handler.cc

```diff
@@ -2047,16 +2055,17 @@ namespace hp
     if (fe_collection.size() > 0)
       {
         // Unpack active_fe_indices.
-        active_fe_indices.resize(tria->n_active_cells(),
-                                 numbers::invalid_unsigned_int);
-        cell_data_transfer->unpack(active_fe_indices);
+        active_fe_index_transfer.active_fe_indices.resize(
+          tria->n_active_cells(), numbers::invalid_unsigned_int);
+        active_fe_index_transfer.cell_data_transfer->unpack(
+          active_fe_index_transfer.active_fe_indices);
 
         // Update all locally owned active_fe_indices.
-        set_active_fe_indices(active_fe_indices);
+        set_active_fe_indices(active_fe_index_transfer.active_fe_indices);
 
         // Free some memory.
-        active_fe_indices.clear();
-        active_fe_indices.shrink_to_fit();
+        active_fe_index_transfer.active_fe_indices.clear();
+        active_fe_index_transfer.active_fe_indices.shrink_to_fit();
```

Comment:
and I think this would be the place where you would simply delete the object (i.e., do `active_fe_index_transfer.reset()`).